src/systems/shadow.js

```diff
@@ -44,10 +41,25 @@ module.exports.System = registerSystem('shadow', {
    * @param {boolean} enabled
    */
   setShadowMapEnabled: function (enabled) {
+    var sceneEl = this.sceneEl;
     var renderer = this.sceneEl.renderer;
-    this.shadowMapEnabled = this.data.enabled && enabled;
-    if (renderer) {
-      renderer.shadowMap.enabled = this.shadowMapEnabled;
+
+    this.shadowMapEnabled = enabled;
+    var newEnabledState = this.data.enabled && this.shadowMapEnabled;
+    if (renderer && newEnabledState !== renderer.shadowMap.enabled) {
+      renderer.shadowMap.enabled = newEnabledState;
+
+      // If scene has already rendered, materials must be updated.
```

Comment:
Maybe this can be factored out to an independent function ([example](https://github.com/aframevr/aframe/blob/master/src/core/a-entity.js#L859)) at the bottom of the file. `updateAllMaterials(sceneEl)`

Reply:
Done